File: src/main/java/superstitio/cards/lupa/SkillCard/ExposeSelf.java
package superstitio.cards.lupa.SkillCard;

import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import superstitio.DataManager;
import superstitio.cards.lupa.LupaCard;
import superstitio.delayHpLose.RemoveDelayHpLoseBlock;

/**
 * 抵消敌人的攻击，转换为精液
 */
public class ExposeSelf extends LupaCard {
    public static final String ID = DataManager.MakeTextID(ExposeSelf.class);

    public static final CardType CARD_TYPE = CardType.SKILL;

    public static final CardRarity CARD_RARITY = CardRarity.COMMON;

    public static final CardTarget CARD_TARGET = CardTarget.SELF;

    private static final int COST = 1;
    private static final int BLOCK = 6;
    private static final int UPGRADE_BLOCK = 3;

    public ExposeSelf() {
        super(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET);
        this.setupBlock(BLOCK, UPGRADE_BLOCK, new RemoveDelayHpLoseBlock());
    }

    @Override
    public void use(AbstractPlayer player, AbstractMonster monster) {
        addToBot_gainBlock();
        addToBot_applyPower(new superstitio.powers.lupaOnly.ExposeSelf(player, 1));
    }

    @Override
    public void upgradeAuto() {
    }
}

File: superstitioMain/java/superstitio/cards/lupa/SkillCard/getSemen/ExposeSelf.kt
package superstitio.cards.lupa.SkillCard.getSemen

import com.megacrit.cardcrawl.actions.common.ApplyPowerAction
import com.megacrit.cardcrawl.cards.DamageInfo
import com.megacrit.cardcrawl.cards.DamageInfo.DamageType
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.core.AbstractCreature
import com.megacrit.cardcrawl.dungeons.AbstractDungeon
import com.megacrit.cardcrawl.monsters.AbstractMonster
import superstitio.DataManager
import superstitio.cards.lupa.LupaCard
import superstitio.cards.lupa.SkillCard.getSemen.ExposeSelf.ExposeSelfPower
import superstitio.delayHpLose.RemoveDelayHpLoseBlock
import superstitio.powers.AbstractSuperstitioPower
import superstitio.powers.lupaOnly.OutsideSemen
import superstitioapi.utils.addToBot_AutoRemoveOne

/**
 * 抵消敌人的攻击，转换为精液
 */
class ExposeSelf : LupaCard(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET)
{
    init
    {
        this.setupBlock(BLOCK, UPGRADE_BLOCK, RemoveDelayHpLoseBlock())
    }

    override fun use(player: AbstractPlayer?, monster: AbstractMonster?)
    {
        addToBot_gainBlock()
        addToBot_applyPower(ExposeSelfPower(AbstractDungeon.player, 1))
    }

    override fun upgradeAuto()
    {
    }

    class ExposeSelfPower(owner: AbstractCreature, amount: Int) : AbstractSuperstitioPower(POWER_ID, owner, amount)
    {
        override fun onAttacked(info: DamageInfo, damageAmount: Int): Int
        {
            if (info.type == DamageType.NORMAL)
            {
                this.flash()
                this.addToBot(
                    ApplyPowerAction(
                        AbstractDungeon.player,
                        AbstractDungeon.player,
                        OutsideSemen(AbstractDungeon.player, damageAmount / 2)
                    )
                )
            }
            return super.onAttacked(info, damageAmount)
        }

        override fun atStartOfTurn()
        {
            addToBot_AutoRemoveOne(this)
        }

        override fun updateDescriptionArgs()
        {
        }

        companion object
        {
            val POWER_ID: String = DataManager.MakeTextID(ExposeSelfPower::class.java)
        }
    }

    companion object
    {
        val ID: String = DataManager.MakeTextID(ExposeSelf::class.java)

        val CARD_TYPE: CardType = CardType.SKILL

        val CARD_RARITY: CardRarity = CardRarity.UNCOMMON

        val CARD_TARGET: CardTarget = CardTarget.SELF

        private const val COST = 1
        private const val BLOCK = 9
        private const val UPGRADE_BLOCK = 3
    }
}
